resources/views/product/partials/gallery/thumbnails.blade.php

```diff
@@ -1,40 +1,49 @@
-@php($breakpoints = ['xl' => 7, 'lg' => 5, 'md' => 4, 'sm' => 3, 'xs' => 4])
-<div v-if="images.length > 1" class="flex mt-3 gap-2">
-    <button
-        v-for="(image, imageId) in images.slice(0, {{ max($breakpoints) }})"
-        class="flex items-center justify-center bg-white border rounded p-1.5 aspect-square max-w-24 flex-1 transition-all outline-transparent overflow-hidden relative"
-        :class="{
-            'outline outline-1 !outline-primary border-primary': active == imageId,
-            'xl:hidden': imageId > {{ $breakpoints['xl'] }},
-            'lg:max-xl:hidden': imageId > {{ $breakpoints['lg'] }},
-            'md:max-lg:hidden': imageId > {{ $breakpoints['md'] }},
-            'sm:max-md:hidden': imageId > {{ $breakpoints['sm'] }},
-            'max-sm:hidden': imageId > {{ $breakpoints['xs'] }}
-        }"
-        @click="change(imageId)"
-    >
-        <img
-            :src="'/storage/{{ config('rapidez.store') }}/resizes/80x80/magento/catalog/product' + image + '.webp'"
-            alt="{{ $product->name }}"
-            class="object-contain block w-auto max-h-full"
-            width="80"
-            height="80"
-        />
-        <span
-            v-if="(imageId + 1) < images.length"
-            v-on:click="toggleZoom()"
-            class="absolute inset-0 items-center justify-center bg-black/20 hidden"
-            :class="{
-                'xl:flex': imageId === {{ $breakpoints['xl'] }},
-                'lg:max-xl:flex': imageId === {{ $breakpoints['lg'] }},
-                'md:max-lg:flex': imageId === {{ $breakpoints['md'] }},
-                'sm:max-md:flex': imageId === {{ $breakpoints['sm'] }},
-                'max-sm:flex': imageId === {{ $breakpoints['xs'] }}
+{{--
+    In this component we avoid layout shifts on the page by pre-populating the possible amount of thumbnail slots.
+    On page load all of the images come from PHP, using `@unless (...) v-cloak @endunless` to hide things that
+    shouldn't be shown, but need a placeholder element.
+    After Vue gets loaded in, it will take over and replace the `v-cloak`s with the `v-if`s, which means that
+    the two conditionals should be identical to avoid layout shifts.
+
+    The `$breakpoints` variable determines how many images are shown on each individual breakpoint.
+--}}
+
+@php($breakpoints = ['xl' => 7, 'lg' => 5, 'md' => 7, 'sm' => 5, 'xs' => 3])
+
+<div class="mt-3 flex gap-2">
+    @for ($imageId = 0; $imageId < max($breakpoints); $imageId++)
+        <button
+            @unless ($imageId < count($product->images)) v-cloak @endunless
```

Comment:
Maybe [`@attributes`](https://github.com/rapidez/blade-directives#attributes) could make this cleaner?

Reply:
```blade
@unless ($imageId < count($product->images)) v-cloak @endunless
v-if="{{ $imageId }} < images.length"
```

vs

```blade
@attributes(['v-cloak' => !($imageId < count($product->images))])
v-if="{{ $imageId }} < images.length"
```

vs

```blade
@attributes([
    'v-cloak' => !($imageId < count($product->images)),
    'v-if' => $imageId . ' < images.length',
])
```

To me the first one (original) is still the most readable. I definitely see the value in using something like `@attributes` for this but to me it feels like the code gets less readable by making it a bit more abstract.

We could maybe make a specific directive just for this sort of functionality, where you want the v-cloak and v-if combined, don't know if that's really worthwhile though.

Reply:
Option 3 feels the cleanest to me, maybe ask some more?

With option 1:
- We don't use `@unless` that much
- The `v-cloak` is somewhere on that line; even with syntax highlighting you don't spot it directly

With option 3:
- You see directly all attributes
- The condition is readable

Reply:
I've asked the team and it looks like we have a majority vote for the 3rd option, so I'll swap it for that.